mlflow/sklearn.py

```diff
@@ -424,3 +424,46 @@ def load_model(model_uri):
     return _load_model_from_local_file(
         path=sklearn_model_artifacts_path, serialization_format=serialization_format
     )
+
+
+# NOTE: The current implementation doesn't guarantee thread-safety, but that's okay for now because:
+# 1. We don't currently have any use cases for allow_children=True.
+# 2. The list append & pop operations are thread-safe, so we will always clear the session stack
+#    once all _SklearnTrainingSessions exit.
+class _SklearnTrainingSession(object):
+    _session_stack = []
+
+    def __init__(self, clazz, allow_children=True):
+        """
+        A session manager for nested autologging runs.
+
+        :param clazz: A class object that this session originates from.
+        :param allow_children: If True, allows autologging in child sessions.
+                               If False, disallows autologging in all descendant sessions.
```

Comment:
@dbczumar What do you think about providing some examples here like below?

```python

>>> with _SklearnTrainingSession(Parent, False) as p:
...     with _SklearnTrainingSession(Child, True) as c:
...         with _SklearnTrainingSession(Grandchild, True) as g:
...             print(p.should_log())  # -> True
...             print(c.should_log())  # -> False
...             print(g.should_log())  # -> False

>>> with _SklearnTrainingSession(Parent, True) as p:
...    with _SklearnTrainingSession(Child, False) as c:
...        with _SklearnTrainingSession(Grandchild, True) as g:
...            print(p.should_log())  # -> True
...            print(c.should_log())  # -> True
...            print(g.should_log())  # -> False
```

Reply:
Sure! An example like this seems valuable to include